src/app/pages/wallets/wallet-details/pages-wallet-details-item.model.ts

```diff
@@ -1,17 +1,34 @@
-import { IWalletTransactionApiResponse, WalletTransactionType } from 'src/app/domains/wallets/domains.wallets.types';
+import { TServerEntityId } from 'src/app/common/http/common.http.types';
+import {
+  WalletTransactionType,
+} from 'src/app/domains/transactions/domains.transactions.constants';
+import {
+  IWalletTransactionApiResponse,
+  IWalletTransactionItemData,
+} from 'src/app/domains/transactions/domains.transactions.types';
 
 export class WalletsDetailsTransaction {
-  public readonly id: number | null;
+  public readonly id: TServerEntityId | null;
   public readonly date: Date;
   public readonly description: string | null;
   public readonly type: WalletTransactionType;
   public readonly amount: number;
 
-  public constructor(data: IWalletTransactionApiResponse) {
+  public constructor(data: IWalletTransactionItemData) {
     this.id = data.id;
     this.date = new Date(data.creationDate);
     this.description = data.description ?? null;
     this.type = data.type;
     this.amount = data.amount;
   }
+
+  public static create(data: Partial<IWalletTransactionApiResponse>): WalletsDetailsTransaction {
+    return new WalletsDetailsTransaction({
+      id: data.id ?? null,
+      amount: data.amount ?? 0,
+      creationDate: data.creationDate ?? '',
+      description: data.description ?? '',
```

Comment:
`description` can be null, so instead of empty string it should be `null` as default

Reply:
Changed for undefined, looks better. Description will be created in constructor as null out of '' or undefined